test/PackUnPackTest.cc

```diff
@@ -0,0 +1,678 @@
+    AreaSize(64, 128), AreaSize(128, 64)};
+AreaSize TEST_AVG_SIZES[] = {AreaSize(4, 4),
+                             AreaSize(4, 8),
+                             AreaSize(8, 4),
+                             AreaSize(8, 8),
+                             AreaSize(16, 16),
+                             AreaSize(4, 16),
+                             AreaSize(16, 4),
+                             AreaSize(16, 8),
+                             AreaSize(8, 16),
+                             AreaSize(32, 32),
+                             AreaSize(32, 8),
+                             AreaSize(16, 32),
+                             AreaSize(8, 32),
+                             AreaSize(32, 16),
+                             AreaSize(16, 64),
+                             AreaSize(64, 16),
+                             AreaSize(64, 64),
+                             AreaSize(64, 32),
+                             AreaSize(32, 64)};
+
+class PackTest : public ::testing::Test,
+                 public ::testing::WithParamInterface<AreaSize> {
+  public:
+    PackTest()
+        : area_width_(std::get<0>(GetParam())),
+          area_height_(std::get<1>(GetParam())) {
+        in_stride_ = out_stride_ = MAX_SB_SIZE;
+        test_size_ = MAX_SB_SQUARE;
+    }
+
+    void SetUp() override {
+        in_compn_bit_buffer1_ =
+            reinterpret_cast<uint8_t *>(eb_aom_memalign(32, test_size_));
+        in_compn_bit_buffer2_ =
+            reinterpret_cast<uint8_t *>(eb_aom_memalign(32, test_size_));
+        in_bit_buffer_ =
+            reinterpret_cast<uint8_t *>(eb_aom_memalign(32, test_size_));
+        local_cache1_ =
+            reinterpret_cast<uint8_t *>(eb_aom_memalign(32, test_size_));
+        local_cache2_ =
+            reinterpret_cast<uint8_t *>(eb_aom_memalign(32, test_size_));
+        memset(in_compn_bit_buffer1_,
+               0,
+               test_size_ * sizeof(in_compn_bit_buffer1_[0]));
+        memset(in_compn_bit_buffer2_,
+               0,
+               test_size_ * sizeof(in_compn_bit_buffer2_[0]));
+        memset(local_cache1_, 0, test_size_ * sizeof(local_cache1_[0]));
+        memset(local_cache2_, 0, test_size_ * sizeof(local_cache2_[0]));
+    }
+
+    void TearDown() override {
+        if (in_bit_buffer_)
+            eb_aom_free(in_bit_buffer_);
+        if (in_compn_bit_buffer1_)
+            eb_aom_free(in_compn_bit_buffer1_);
+        if (in_compn_bit_buffer2_)
+            eb_aom_free(in_compn_bit_buffer2_);
+        if (local_cache1_)
+            eb_aom_free(local_cache1_);
+        if (local_cache2_)
+            eb_aom_free(local_cache2_);
+    }
+
+  protected:
+    void check_output(uint32_t width, uint32_t height, uint8_t *out_1,
+                      uint8_t *out_2) {
+        int fail_count = 0;
+        for (uint32_t j = 0; j < height; j++) {
+            for (uint32_t k = 0; k < width / 4; k++) {
+                if (out_1[k + j * ((width == 32) ? 8 : 16)] !=
```

Comment:
Seems like the stride of out_1 buffer is fixed to 8/16, whereas the stride of out_2 is fixed to out_stride_. It's not good to put out_1 as function parameter.

Reply:
Using the out_stride to access the output buffer of c_pack_avx2_intrin